enterprise-suite/scripts/lbc.py

```diff
@@ -0,0 +1,462 @@
+    stdout, returncode = run('kubectl version', DEFAULT_TIMEOUT)
+    if returncode != 0:
+        msg = 'Cannot reach cluster with kubectl'
+        if minishift:
+            # Minishift needs special configuration for kubectl to work
+            msg = msg + ". Did you do 'eval $(minishift oc-env)'?"
+        fail(msg)
+
+def check_credentials(creds):
+    registry = 'https://lightbend-docker-commercial-registry.bintray.io/v2'
+    api_url = registry + '/enterprise-suite/es-monitor-api/tags/list'
+
+    # Set up basic auth with given creds
+    req = urllib2.Request(api_url)
+    basic_auth = base64.b64encode('{}:{}'.format(creds[0], creds[1]))
+    req.add_header('Authorization', 'Basic ' + basic_auth)
+
+    success = False
+    try:
+        resp = urllib2.urlopen(req)
+        if resp.getcode() == 200:
+            # Lazy way of verifying returned json - there should be a tag named "latest"
+            if '"latest"' in resp.read():
+                success = True
+    finally:
+        return success 
+
+def preinstall_check(creds, minikube=False, minishift=False):
+    check_helm()
+    check_kubectl()
+
+    if minikube:
+        require_version('minikube version', REQ_VER_MINIKUBE)
+
+    if minishift:
+        require_version('minishift version', REQ_VER_MINISHIFT)
+        require_version('oc version', REQ_VER_OC)
+
+    # Check if helm is set up inside a cluster
+    stdout, returncode = run('helm version', DEFAULT_TIMEOUT)
+    if returncode != 0:
+        fail('Cannot get helm status. Did you set up helm inside your cluster?')
+
+    # TODO: Check if RBAC rules for tiller are set up
+
+    if not check_credentials(creds):
+        fail('Your credentials do not appear to be correct' +
+                 ' - unable to make authenticated request to lightbend docker registry')
+
+def install_helm_chart(creds_file):
+    creds_arg = '--values ' + creds_file
+    # Helm args are separated from lbc.py args by double dash, filter it out
+    helm_args = ' '.join([arg for arg in args.helm if arg != '--'])
+    version_arg = ('--version ' + args.version) if args.version != None else '--devel'
+
+    chart_ref = None
+    if args.local_chart != None:
+        # Install from local chart tarball
+        chart_ref = args.local_chart
+    else:
+        execute('helm repo add es-repo ' + args.repo)
+        execute('helm repo update')
+        chart_ref = 'es-repo/' + args.chart
+    
+    if args.export_yaml != None:
+        # Tilerless path - renders kubernetes resources and prints to stdout
+
+        creds_exec_arg = ''
+        if args.export_yaml == 'creds':
+            creds_exec_arg = '--execute templates/commercial-credentials.yaml ' + creds_arg
+            printerr('warning: credentials in yaml are not encrypted, only base64 encoded. Handle appropriately.')
+        
+        try:
+            tempdir = make_tempdir()
+            execute('helm fetch -d {} {} {}'
+                .format(tempdir, version_arg, chart_ref))
```

Comment:
This has the same bug as install-es.sh that we fixed recently. This shouldn't be run if `--local-chart` is set.

Reply:
Done